### PR TITLE
Update our API files for 2.0.0

### PR DESCRIPTION
**Pull request by jaredpar**
This both updates our API files for 2.0 and creates a tool to automate this process in the future.  That will avoid the hours of time it took me to merge the changes into our present branch. 

The process I took was the following:

- Wrote the automated script
- Sync'd to dev15.0.x, ran the script and checked in the update 
- Sync'd to master and cherry picked the change above
- Did a build and cleaned up all of the new APIs added since 

This would've been a lot smoother with fix all in solution functioning 😦

### CODE REVIEW

**Comment by jaredpar** [comment]
CC @sharwell, @jasonmalinowski, @DustinCampbell, @gafter, @VSadov, @AlekseyTs

**Comment by jasonmalinowski** [comment]
@jaredpar I would support making this change in dev15.2.x so we don't regress stuff there. I have no idea if this is hell or not to retarget.

**Comment by jaredpar** [comment]
>  I would support making this change in dev15.2.x so we don't regress stuff there. I have no idea if this is hell or not to retarget.

It is unfortunately.  The IDE is currently broken for "Fix All in Solution" and hangs about 1/2 following "Fix All in Project" then Build.

**Comment by jasonmalinowski** [comment]
@jaredpar Do we have bugs tracking those issues?

**Comment by jaredpar** [comment]
@jasonmalinowski 

My understanding is the Fix All in Solution is known.  The other I don't have a repo or stack yet.  Once I get one I will file it.

**Comment by jasonmalinowski** [comment]
@jaredpar Please file bugs, even if you think it's already known. Better to confirm a dupe than have too many cooks in the kitchen.

**Comment by jaredpar** [comment]
This is the fix all in solution bug https://github.com/dotnet/roslyn/issues/18124

File: src/Tools/PublicApi/mark-shipped.ps1 [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        if ($item.StartsWith($removedPrefix)) {
+            $item = $item.Substring($removedPrefix.Length)
+            $removed += $item
+        }
+        else {
+            $shipped += $item
+        }
+    }
+
+    $shipped | ?{ -not $removed.Contains($_) } |  Out-File $shippedFilePath -Encoding Ascii
```
Comment (sharwell) [inline src/Tools/PublicApi/mark-shipped.ps1]: :question: Shouldn't this sort the result?

Comment (jaredpar) [inline src/Tools/PublicApi/mark-shipped.ps1]: Ideally yes.  Practically it causes merge hell because our code fixer doesn't sort.

Comment (sharwell) [inline src/Tools/PublicApi/mark-shipped.ps1]: I thought the code fixer did sort in some manner. I noticed additions do not always get added just at the end.

Comment (jaredpar) [inline src/Tools/PublicApi/mark-shipped.ps1]: Hmm looks like that was the intent of the fixer.  But in practice there were cases where it wasn't sorted

https://github.com/dotnet/roslyn-analyzers/blob/master/src/Roslyn.Diagnostics.Analyzers/Core/DeclarePublicAPIFix.cs#L69

**Comment by jaredpar** [comment]
@sharwell changed everything to be sorted.  

It feels odd to have people remind me to implement sorting sanity given that I'm the person who originally added the sorting sanity because it made me feel more sane.  😄

**Comment by sharwell** [comment]
@jaredpar For the purpose of clarity, can you identify the specific commit which is considered public for the purpose of creating 23cfeea?

**Comment by jaredpar** [comment]
@sharwell the source commit for the API cut off is  0449c8c4fda150579ea8a573459ba48d566fbda2

**Pull request merged**